### PR TITLE
General Use Wallet Examples

### LINKED ISSUE #432: Documentation requires general use examples
General use examples are required for explicit referencing to meet the needs of #416

### PR DESCRIPTION
**Pull request by dskvr**
Many documentation cases are presently included inside the Smart Contracts walkthrough, and because of this cannot be externally linked in an explicit manner. Consensus was reached in telegram chat favoring temporary redundancy in `readme.md` as opposed to moving the walkthrough to a tutorials section or linking externally to contextually irrelevant walkthrough steps and forcing the user to read between the lines.

Originally included were general use contract functions but removed them from this branch, as this section needs more information but lack in priority comparatively. 

Missing are instructions for permissions. 

Resolves #432

### CODE REVIEW

**Comment by dskvr** [comment]
I would like to squash these commits before they are merged.

File: README.md [hunk 1]
```diff
@@ -388,58 +397,54 @@ cd ~/eos/build/programs/eosd/
...
+As you've previously added `plugin = eos::wallet_api_plugin` into `config.ini`, EOS wallet will be running as a part of `eosd` process. Every contract requires an associated account, so first, create a wallet.
 
 ```bash
 cd ~/eos/build/programs/eosc/
 ./eosc wallet create # Outputs a password that you need to save to be able to lock/unlock the wallet
-./eosc wallet import 5KQwrPbwdL6PhXujxW37FSSQZ1JiwsST4cqQzDeyXtP79zkvFD3
 ```
 
-Now you can issue API commands under `inita` authority.
+For the purpose of this walkthrough, we'll import the private key of the `inita` account so that you can issue API commands under authority of an existing account. This private key can be found within your `config.ini` and is provided for testing purposes.
```
Comment (sergmetelin) [inline README.md]: Should we say that 'inita' is a pre-created account from genesis file?

File: README.md [hunk 2]
```diff
@@ -592,43 +588,107 @@ You can find more detailed API documentation in Doxygen reference: https://eosio
...
+<a name="generaluse"></a>
+## General Use Examples
 
-Build eos image:
+Below are a list of general purpose examples
 
+<a name="generaluse-wallets"></a>
+### Wallets
+
+Wallets may contain several accounts, and each account is has various permissions for different levels of authentication. Wallets are either in a locked or unlocked state.
```
Comment (sergmetelin) [inline README.md]: is has

File: README.md [hunk 3]
```diff
@@ -592,43 +588,107 @@ You can find more detailed API documentation in Doxygen reference: https://eosio
...
+<a name="generaluse-generatekeys"></a>
+#### Generate Keys
+
+The below example generates a public/private key pair for the `owner_key` and `active_key` permissions. 
+
+```bash
+./eosc create key
+```
+
+Each time you `create key` you wil recieve a keypair.
```
Comment (sergmetelin) [inline README.md]: recieve -> receive

File: README.md [hunk 4]
```diff
@@ -592,43 +588,107 @@ You can find more detailed API documentation in Doxygen reference: https://eosio
...
+./eosc create account inita bob owner_public_key active_public_key 
 ```
 
-Get chain info:
+You should get a json response back with a transaction ID confirming it was executed successfully.
+
+<a name="generaluse-accountexists"></a>
+#### Checking Existence of Account
+
+Check that account an account exists or was created successfully.
```
Comment (sergmetelin) [inline README.md]: typo: account an account

File: README.md [hunk 5]
```diff
@@ -592,43 +588,107 @@ You can find more detailed API documentation in Doxygen reference: https://eosio
...
 ```bash
-docker exec docker_eos_1 eosc set contract exchange build/contracts/exchange/exchange.wast build/contracts/exchange/exchange.abi
+./eosc wallet import 5XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 ```
+
+The **bob** account is now imported into your wallet
+
+
+<a name="generaluse-pushmessage"></a>
+#### Push message to Contract
```
Comment (sergmetelin) [inline README.md]: To maintain style - pushing

File: README.md [hunk 6]
```diff
@@ -592,43 +588,107 @@ You can find more detailed API documentation in Doxygen reference: https://eosio
...
+
+To send a message to a contract, we need to specify the **contract** and its **method** we are pushing the message to, the **parameters** anticipated by the contract's method and then some additional arguments depending on the nature of the message. 
+
+The below example pushes a message to the **transfer** method of the sample **currency** contract from user **bob**, it will include parameters that specify the from (*bob*) and to (*inita*) account, and the amount to transfer (*50*), we'll set the `--scope` of the message to include these two accounts `bob,inita`, and the `--permissions` required to send the message, which are bob's active permissions, expressed with `bob@active`.
+
+```./eosc push message grocery transfer '{"from":"grocery","to":"bob","amount":50}' --scope grocery,inita --permission grocery@activex```
```
Comment (sergmetelin) [inline README.md]: activex -> active

Comment (sergmetelin) [reply README.md]: also, shouldn't it be push message currency transfer?

Comment (sergmetelin) [reply README.md]: and is the scope right?

Comment (dskvr) [reply README.md]: That's an artifact from another branch, thought that wasn't included.

File: README.md (hunk 1 above)
Comment (dskvr) [reply README.md]: That's not related to this issue, I'll commit to `docs-434` and merge into here.

**Comment by dskvr** [comment]
@sergmetelin Updated

File: README.md (hunk 1 above)
Comment (dskvr) [reply README.md]: Additionally, there's actually instructions in here from before modification to create `inita` account that I noticed but didn't have time to address

Comment (dskvr) [reply README.md]: Ok, I see the problem. The tester account was initially used in this example to demonstrate creating a wallet,  but the convenience of the existing `inita` account was never removed. Then there was a mistake where `inita` was referenced for create wallet.

**Comment by dskvr** [comment]
Looks like this already exists in [Doxygen](https://eosio.github.io/eos/group__eosc.html#createwallet)

**Pull request closed without merging**